File: python/stdlib/default/function.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("Function", "default.function", {"code": 'result = value["val"]'},
                                   {"val": "Object"},
                                   {"result": "Object"},
                                   "Executes the argument as pyton code.", verbose)
        self.args = args

    def tick(self, value):
        result = {}
        tag = None
        if "tags" in value and "val" in value["tags"]:
            tag = value["tags"]["val"]
        
        exec(self.args["code"])
        
        if tag:
            return {"result": result, "tags":{"result":tag}}
        else:
            return {"result": result}

File: lua/stdlib/default/function.lua
local myNode = {}
function myNode.init(node)
  node.tick = function(value)
    result = {}
    result.result = value.val
    print("Not implemented yet: " .. node.code)
    return result
  end
end

function myNode.spec(node)
  node.name = "Function"
  node.inputs.val = "Object"
  node.outputs.result = "Object"
  node.args.code = "result.result = value.val"
  node.desc = "Execute a custom function"
end

return myNode
